app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/MainActivity.java

```diff
@@ -143,28 +142,21 @@ protected void onCreate(Bundle savedInstanceState) {
         SharedPreferences sharedPref = getSharedPreferences("ExpansionFile", MODE_PRIVATE);
         int defaultFileVersion = 0;
 
-        String flagFilePath = "/storage/emulated/0/Android/data/com.maq.xprize.onecourse.hindi/files/.success.txt";
-        File flagFile = new File(flagFilePath);
-        if (!flagFile.exists()) {
+        // Retrieve the stored values of main and patch file version
+        int storedMainFileVersion = sharedPref.getInt(getString(R.string.mainFileVersion), defaultFileVersion);
+        int storedPatchFileVersion = sharedPref.getInt(getString(R.string.patchFileVersion), defaultFileVersion);
+        boolean isExtractionRequired = isExpansionExtractionRequired(storedMainFileVersion, storedPatchFileVersion);
+
+        if (storedMainFileVersion == 0 && storedPatchFileVersion == 0) {
             // Set main and patch file version to 0, if the extractions takes place for the first time
             SharedPreferences.Editor editor = sharedPref.edit();
             editor.putInt(getString(R.string.mainFileVersion), 0);
             editor.putInt(getString(R.string.patchFileVersion), 0);
             editor.commit();
```

Comment:
Use editor.apply()

Reply:
Since the splash screen activity retrieves the content immediately, using `editor.commit()` is the best option rather than hoping for the changes to get applied before they are accessed.